src/estimation/src/AttitudeMahonyFilter.cpp

```diff
@@ -95,7 +123,7 @@ bool iDynTree::AttitudeMahonyFilter::propagateStates()
 
     // system dynamics equations
     q = q + (dq*(m_params.time_step_in_seconds*0.5));
-    if (q.norm() != 0)
+    if (q.norm() != 0 || q.norm() != 1)
```

Comment:
This should be `&&` condition.

Reply:
I am confused. How can the norm be both 0.0 and 1.0 ?

Reply:
Normalize only if the norm is not zero (to avoid nan) and the norm is not one(to become unit quaternion if already not)

Reply:
Does it make sense ? Or should we actually split it into two conditions ?

Reply:
Why the quaternion should be ever 0.0?

Reply:
Because we only typedef UnitQuaternion and not type check. We don't establish anywhere that the vector will have unit norm or not have zero norm at all.
Worst case check!

Reply:
If the quaternion norm is zero, then an error should be emitted at some point, and the `propagateStates` method shoul return `false`, right?

Reply:
True. Makes sense. I will change the condition.

Reply:
@traversaro fixed this condition and improved the test.